src/libsync/discoveryphase.cpp

```diff
@@ -299,12 +304,14 @@ void DiscoverySingleDirectoryJob::directoryListingIteratedSlot(QString file, con
 {
     //qDebug() << Q_FUNC_INFO << _subPath << file << map.count() << map.keys() << _account->davPath() << _lsColJob->reply()->request().url().path();
     if (!_ignoredFirst) {
-        // First result is the directory itself. Maybe should have a better check for that? FIXME
+        // The first entry is for the folder itself, we should process it differently.
         _ignoredFirst = true;
         if (map.contains("permissions")) {
             emit firstDirectoryPermissions(map.value("permissions"));
         }
-
+        if (true || map.contains("data-fingerprint")) {
```

Comment:
I do not like the isRootPath parameter. It's cluttering of the interface.

I think it would be nicer if you would either set all props from the class that creates the job using `setProperties()`, or even better add a method `addProperty( a_property)` that is called in case it is a root object.